gdal/frmts/mrf/marfa_dataset.cpp

```diff
@@ -149,8 +149,13 @@ int MRFDataset::CloseDependentDatasets()
 MRFDataset::~MRFDataset()
 
 {   // Make sure everything gets written
-    MRFDataset::FlushCache();
+    if (eAccess != GA_ReadOnly && !bCrystalized)
+        if (!Crystalize()) {
+            // Can't return from a destructor, just flag the error
```

Comment:
this comment is a bit inaccurate. You can definitely have a return statement in a destructor (not saying this might be appropriate to put one here, if they are other memory cleanups to do)

Reply:
I meant return something, like an error flag

Reply:
```suggestion
            // Can't return error code from a destructor, just emit the error
```